src/Support/Laravel/ServiceProvider.php

```diff
@@ -53,24 +53,30 @@ public function register()
             /** @var \DebugBar\DebugBar $debugBar */
             $debugBar = $this->app->make('debugbar');
 
-            $stack->push(new Middleware(new Profiler($timeline = $debugBar->getCollector('time'))));
-            $stack->unshift(new ExceptionMiddleware($debugBar->getCollector('exceptions')));
+            if (
+              $debugBar->hasCollector('exceptions') &&
+              $debugBar->hasCollector('messages') &&
+              $debugBar->hasCollector('time')
```

Comment:
why check all 3 at the same time? Why not per collector and then add a specific middleware to the stack or not.

Reply:
Fair point. Gonna update.